app/src/main/java/com/duckduckgo/app/tabs/db/TabsDao.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.tabs.db
+
+import android.arch.lifecycle.LiveData
+import android.arch.persistence.room.*
+import com.duckduckgo.app.tabs.model.TabEntity
+import com.duckduckgo.app.tabs.model.TabSelectionEntity
+import javax.inject.Singleton
+
+@Dao
+@Singleton
+abstract class TabsDao {
+
+    @Query("select * from tabs limit 1")
+    abstract fun firstTab(): TabEntity?
+
+    @Query("select tabs.tabId, url, title from tabs inner join tab_selection ON tabs.tabId = tab_selection.tabId limit 1")
```

Comment:
I think some of this complexity could be reduced using the `@Relation` annotation but we can revisit later since it is working as is. https://developer.android.com/reference/android/arch/persistence/room/Relation.html

Reply:
> The type of the field annotated with Relation must be a List or Set
I felt a query was easier than the Relation restrictions for what I wanted here. Would be interested in hearing more if there's a cleaner way. Let's zoom on it soon.